src/mon/MonmapMonitor.cc

```diff
@@ -808,3 +830,118 @@ void MonmapMonitor::check_sub(Subscription *sub)
     }
   }
 }
+
+namespace ceph::mon_cmds {
+
+using std::string;
+using std::vector;
+
+using ceph::mon_cmds::mon_cmd;
+using ceph::mon_cmds::mon_cmd_result;
+
+// A demo type for testing hooks:
+#pragma warning JFW REMOVEME
```

Comment:
Here's a simple example of adding a new command that binds to "ceph mon echo" at both prepare and preprocess phases.